src/components/Items/Items.js

```diff
@@ -92,18 +108,66 @@ export class Items extends Component {
       storeBlock = (
         <h3>
           {store.name}
+
           {address}
         </h3>
       );
     }
+    let header;
+    let editForm = (
+      <form onSubmit={e => this.newName(e)}>
+        <input
+          type="text"
+          ref={editListName => {
+            this.editListName = editListName;
+          }}
+        />
+        <button>
+          <img
+            className="editIcon"
+            src="/edit.png"
+            alt="editList"
+            type="submit"
+          />
+        </button>
+      </form>
+    );
+    if (editingName) {
```

Comment:
I think you could rewrite this large if block (with lots of repeated code) as something like this:

```javascript
header= (
  <header...>
    <h1>
      { editingName ? editForm : name }
    </h1>
    ...
  </header>
);